### PR TITLE
core: reaction: fix alias handling

### PR DESCRIPTION
**Pull request by mkviatkovskii**

### CODE REVIEW

File: core/indigo-core/tests/common.cpp [hunk 1]
```diff
@@ -75,3 +77,40 @@ std::string IndigoCoreTest::dataPath(const char* dataPathSuffix)
...
+}
+
+void IndigoCoreTest::loadQueryReaction(const char* buf, QueryReaction& queryReaction)
+{
+    BufferScanner scanner(buf);
+    ReactionAutoLoader loader(scanner);
+    loader.loadQueryReaction(queryReaction);
+}
+
+std::string IndigoCoreTest::saveReaction(Reaction& reaction)
```
Comment (AlexanderSavelyev) [inline core/indigo-core/tests/common.cpp:95]: rename to saveReactionSmiles

File: core/indigo-core/tests/common.cpp [hunk 2]
```diff
@@ -75,3 +77,40 @@ std::string IndigoCoreTest::dataPath(const char* dataPathSuffix)
...
+    Array<char> buffer;
+    ArrayOutput output(buffer);
+    RSmilesSaver smilesSaver(output);
+    smilesSaver.saveReaction(reaction);
+    output.writeChar(0);
+    std::string result(buffer.ptr());
+    return result;
+}
+
+std::string IndigoCoreTest::saveQueryReaction(QueryReaction& queryReaction, const bool smarts)
```
Comment (AlexanderSavelyev) [inline core/indigo-core/tests/common.cpp:106]: can we have the same saveReactionSmiles (function overloading)?

File: core/indigo-core/tests/common.cpp [hunk 3]
```diff
@@ -75,3 +77,40 @@ std::string IndigoCoreTest::dataPath(const char* dataPathSuffix)
...
 }
+
+void IndigoCoreTest::loadReaction(const char* buf, Reaction& reaction)
+{
+    BufferScanner scanner(buf);
+    ReactionAutoLoader loader(scanner);
+    loader.loadReaction(reaction);
+}
+
+void IndigoCoreTest::loadQueryReaction(const char* buf, QueryReaction& queryReaction)
```
Comment (AlexanderSavelyev) [inline core/indigo-core/tests/common.cpp:88]: can we have function overloading with loadReaction?

File: core/indigo-core/tests/tests/reaction.cpp [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+class IndigoCoreReactionTest : public IndigoCoreTest
+{
+};
+
+TEST_F(IndigoCoreReactionTest, pseudoatoms)
+{
+    Reaction reaction;
+    loadReaction("*>>* |$Carbon;$|", reaction);
+    ASSERT_STREQ("*>>* |$Carbon;A$|", saveReaction(reaction).c_str());
```
Comment (AlexanderSavelyev) [inline core/indigo-core/tests/tests/reaction.cpp:37]: add more tests for the aliases to verify that reaction molecule has defined alias

File: core/indigo-core/tests/tests/reaction.cpp [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+}
+
+TEST_F(IndigoCoreReactionTest, aliases)
+{
+    Reaction reaction;
+    loadReaction("C>>N |$Carbon;$|", reaction);
+    ASSERT_STREQ("C>>N |$Carbon;$|", saveReaction(reaction).c_str());
```
Comment (AlexanderSavelyev) [inline core/indigo-core/tests/tests/reaction.cpp:44]: add more tests to verify correct alias (use new getAlias from molecule)

File: core/indigo-core/tests/tests/reaction.cpp [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+}
+
+TEST_F(IndigoCoreReactionTest, aliases_complex)
+{
+    QueryReaction reaction;
+    loadQueryReaction("[#6:1]=[#6:2][#6:3].[#6:4]=[#6:5][#6:6]>>[#6:3][#6:2]=[#6:5][#6:6] |$;;R1;;;R2;R1;;;R2$|", reaction);
+    reaction.clearAAM();
+    ASSERT_STREQ("[#6]=[#6]-[#6].[#6]=[#6]-[#6]>>[#6]-[#6]=[#6]-[#6] |$;;R1;;;R2;R1;;;R2$|", saveQueryReaction(reaction, true).c_str());
```
Comment (AlexanderSavelyev) [inline core/indigo-core/tests/tests/reaction.cpp:52]: add more tests for aliases (no just save to smiles)

**Review by AlexanderSavelyev — CHANGES_REQUESTED** [review]

File: core/indigo-core/tests/common.cpp (hunk 1 above)
Comment (mkviatkovskii) [reply core/indigo-core/tests/common.cpp]: Done

File: core/indigo-core/tests/common.cpp (hunk 2 above)
Comment (mkviatkovskii) [reply core/indigo-core/tests/common.cpp]: Done

File: core/indigo-core/tests/common.cpp (hunk 3 above)
Comment (mkviatkovskii) [reply core/indigo-core/tests/common.cpp:91]: Done

File: core/indigo-core/tests/tests/reaction.cpp (hunk 4 above)
Comment (mkviatkovskii) [reply core/indigo-core/tests/tests/reaction.cpp]: Added for CML, JSON and RXNFILE

**Pull request merged**